create-fetchival.js

```diff
@@ -0,0 +1,62 @@
+// Based on https://github.com/typicode/fetchival and aims to keep most of the same API
+
+// Unlike fetchival, we also encode the keys
+function query(params) {
+  if (!params) return ''
+  return `?${Object.entries(params)
+    .map(([k, v]) => `${encodeURIComponent(k)}=${encodeURIComponent(v)}`)
+    .join('&')}`
+}
+
+function createFetchival({ fetch }) {
+  async function _fetch(method, url, opts, data) {
+    // Unlike fetchival, don't silently ignore and override
+    if (opts.body) throw new Error('unexpected pre-set body option')
+
+    // Unlike fetchival, don't pollute the opts object we were given
+    const res = await fetchival.fetch(url, {
+      ...opts,
+      method,
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+        ...(opts.headers || {}),
+      },
+      ...(data ? { body: JSON.stringify(data) } : {}),
+    })
+
+    if (res.status >= 200 && res.status < 300) {
```

Comment:
Could we use `response.ok` here?

Reply:
Should be possible. @ChALkeR any reason why we check the range manually?

Reply:
This is historical code and we keep backwards-compat here.
We can do changes in experimental.

Reply:
Ah, it's ok to change this to `res.ok` as those are identical though. So no specific reason except for inherited code.